src/game/lara/lara_state.c

```diff
@@ -112,7 +115,21 @@ void Lara_State_Run(ITEM_INFO *item, COLL_INFO *coll)
         }
     }
 
-    if (g_Input.jump && !item->gravity_status) {
+    if (g_Config.enable_tr2_jumping) {
+        int16_t anim =
+            item->anim_number - g_Objects[item->object_number].anim_index;
+        if (anim == LA_RUN_START) {
+            jump_permitted = false;
+        } else if (anim == LA_RUN) {
+            if (item->frame_number == LF_JUMP_READY) {
+                jump_permitted = true;
+            }
+        } else {
+            jump_permitted = true;
+        }
+    }
```

Comment:
I believe this boolean logic can be simplified, I'll follow up tomorrow

Reply:
Would this work?
```c
    if (g_Config.enable_tr2_jumping) {
        int16_t anim = item->anim_number - g_Objects[item->object_number].anim_index;
        if (anim == LA_RUN_START) {
            m_JumpPermitted = false;
        } else if (anim != LA_RUN || item->frame_number == LF_JUMP_READY) {
            m_JumpPermitted = true;
        }
    }
```

Reply:
That looks much better - I'll test it out tomorrow. Thanks.

Reply:
An ideal outcome would be to forgo introducing a new global variable and set it always in the local scope. But I don't think that's possible since frame number resets to 0 and cycles, so even comparing it with >= will work in an unintended way.

Reply:
Yes, it is a shame but thankfully this is a short snippet of code now so looks much tidier. I've tested it out and all is good 👍